src/varity/vcf_to_hgvs/protein.clj

```diff
@@ -290,14 +290,21 @@
                            [ppos pref palt])
         [_ _ offset _] (diff-bases pref palt)
         alt-prot-seq (format-alt-prot-seq seq-info)
-        ref (nth (:ref-prot-seq seq-info) (dec (+ ppos offset)))
-        alt (nth alt-prot-seq (dec (+ ppos offset)))
+        [ref alt offset*] (let [original-position (dec (+ ppos offset))]
+                           (loop [ref* (nth (:ref-prot-seq seq-info) original-position)
+                                  alt* (nth alt-prot-seq original-position)
+                                  ex-offset 0]
+                             (if (not= ref* alt*)
+                               [ref* alt* (+ offset ex-offset)]
+                               (recur (nth (:ref-prot-seq seq-info) (+ original-position (inc ex-offset)))
+                                      (nth alt-prot-seq (+ original-position (inc ex-offset)))
+                                      (inc ex-offset)))))
```

Comment:
Thank you for fixing a bug that I missed consideration about trailing alt proteins accidentally corresponded to ref proteins.

Your changed code works correctly, but there is a simpler way; all you have to do is removing `(= pref palt)` condition on L282. The following code passes the test you added.

```clj
  (let [[ppos pref palt] (->> (map vector (:ref-prot-seq seq-info) (:alt-prot-seq seq-info))
                              (drop (dec ppos))
                              (map-indexed vector)
                              (drop-while (fn [[_ [r a]]] (= r a)))
                              first
                              ((fn [[i [r a]]]
                                 [(+ ppos i) (str r) (str a)])))
        [_ _ offset _] (diff-bases pref palt)
```